staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -191,34 +191,36 @@ func (a *Admission) ValidateConfiguration() error {
 // Validate admits an API request.
 // The objects in admission attributes are expected to be external v1 objects that we care about.
 func (a *Admission) Validate(ctx context.Context, attrs Attributes) *admissionv1.AdmissionResponse {
+	logger := newAdmissionLogger(attrs)
+
 	var response *admissionv1.AdmissionResponse
 	switch attrs.GetResource().GroupResource() {
 	case corev1.Resource("namespaces"):
-		response = a.ValidateNamespace(ctx, attrs)
+		response = a.ValidateNamespace(ctx, logger, attrs)
 	case corev1.Resource("pods"):
-		response = a.ValidatePod(ctx, attrs)
+		response = a.ValidatePod(ctx, logger, attrs)
 	default:
-		response = a.ValidatePodController(ctx, attrs)
+		response = a.ValidatePodController(ctx, logger, attrs)
 	}
 
 	// TODO: record metrics.
 
 	return response
 }
 
-func (a *Admission) ValidateNamespace(ctx context.Context, attrs Attributes) *admissionv1.AdmissionResponse {
+func (a *Admission) ValidateNamespace(ctx context.Context, logger *admissionLogger, attrs Attributes) *admissionv1.AdmissionResponse {
```

Comment:
exported method that takes a private type is pretty weird, as is threading both attrs and a logger instance which wraps attrs

would it be cleaner to give `Admission` a handle to a logger and make private helper methods like this:
```
func (a*Admission) logError(attrs Attributes, msg string, keysAndValues ...interface{})
func (a*Admission) logInfo(attrs Attributes, msg string, keysAndValues ...interface{})
func (a*Admission) logVInfo(attrs Attributes, msg string, keysAndValues ...interface{})
```